src/textual/dom.py

```diff
@@ -98,6 +98,9 @@ class DOMNode(MessagePump):
     # True if this node inherits the CSS from the base class.
     _inherit_css: ClassVar[bool] = True
 
+    # True if this node inherits the component classes from the base class.
+    _inherit_component_classes: ClassVar[bool] = True
```

Comment:
The other `inherit_*` values are set from `__init_subclass__`. We should probably do the same for this.